src/test/java/TicketServiceTest.java

```diff
@@ -0,0 +1,211 @@
+    @Mock
+    private TicketRepository ticketRepository;
+    private Ticket ticket;
+
+    @BeforeEach
+    public void setUp() {
+        ticket = mock(Ticket.class);
+    }
+
+    @Test
+    @DisplayName("Should save new ticket")
+    public void saveTicketPositiveTest() {
+        Mockito.when(ticketRepository.save(any(Ticket.class))).thenReturn(ticket);
+
+        ticketService.saveTicket(ticket);
+
+        verify(ticketRepository, times(1)).save(ticket);
+    }
+
+    @Test
+    @DisplayName("Should throw exception when ticket is null")
+    public void saveTicketNegativeTest() {
+        Mockito.when(ticketRepository.save(any(Ticket.class))).thenReturn(null);
+
+        Assertions.assertThrows(RuntimeException.class, () -> {
+            ticketService.saveTicket(null);
+        });
+    }
+
+    @Test
+    @DisplayName("Should throw exception when ticket is empty")
+    public void saveTicketCornerTest() {
+        Mockito.when(ticketRepository.save(any(Ticket.class))).thenThrow(new DataIntegrityViolationException(""));
+
+        Assertions.assertThrows(DataIntegrityViolationException.class, () -> {
+            ticketService.saveTicket(new Ticket());
+        });
+    }
+
+    @Test
+    @DisplayName("Should get all tickets")
+    void getAllTicketsPositiveTest() {
+        Mockito.when(ticketRepository.findAll()).thenReturn(List.of(ticket));
+
+        List<Ticket> actualTickets = ticketService.getAllTickets();
+
+        Assertions.assertEquals(List.of(ticket), actualTickets);
+    }
+
+    @Test
+    @DisplayName("Should throw exception when the returned list is null")
+    void getAllTicketsNegativeTest() {
+        Mockito.when(ticketRepository.findAll()).thenReturn(null);
+
+        List<Ticket> actualTickets = ticketService.getAllTickets();
+
+        assertNull(actualTickets);
+        verify(ticketRepository, times(1)).findAll();
+    }
+
+    @Test
+    @DisplayName("Should return an empty list")
+    void getAllTicketsCornerTest() {
+        Mockito.when(ticketRepository.findAll()).thenReturn(Collections.emptyList());
+
+        List<Ticket> actualTickets = ticketService.getAllTickets();
+
+        assertNotNull(actualTickets);
+        assertTrue(actualTickets.isEmpty());
+        verify(ticketRepository, times(1)).findAll();
+    }
+
+    @Test
+    @DisplayName("Should fetch ticket by id")
+    void fetchTicketByIdPositiveTest() {
+        Mockito.when(ticketRepository.getTicketById(1)).thenReturn(ticket);
+
+        Ticket actualTicket = ticketService.fetchTicketById(1);
+
+        Assertions.assertEquals(ticket, actualTicket);
+        verify(ticketRepository, times(1)).getTicketById(1);
+    }
+
+    @Test
+    @DisplayName("Should return null when ticket wasn't find by id")
+    void fetchTicketByIdNegativeTest() {
+        Mockito.when(ticketRepository.getTicketById(1)).thenReturn(null);
+
+        Ticket actualTicket = ticketService.fetchTicketById(1);
+
+        Assertions.assertNull(actualTicket);
```

Comment:
you could use static import for methods from Assertions class for better readabiliity